app/helpers/sessions_helper.rb

```diff
@@ -0,0 +1,12 @@
+module SessionsHelper
+  # Redirects to stored location (or to the default).
+  def redirect_back_or(default)
+    redirect_to(session[:forwarding_url] || default)
+    session.delete(:forwarding_url)
+  end
+
+  # Stores the URL trying to be accessed.
+  def store_location
+    session[:forwarding_url] = request.referer if request.post?
```

Comment:
No joke... the misspelling made it into the HTTP standard and so now I have to call referer on request.
http://en.wikipedia.org/wiki/HTTP_referer